pathplannerlib/src/main/java/com/pathplanner/lib/path/PathPlannerPath.java

```diff
@@ -259,14 +266,24 @@ private static PathPlannerPath fromJson(JSONObject pathJson) {
       eventMarkers.add(EventMarker.fromJson((JSONObject) markerJson));
     }
 
+    Rotation2d previewStartingRotation = Rotation2d.fromDegrees(0);
+    if (pathJson.containsKey("previewStartingState")) {
+      JSONObject previewStartingStateJson = (JSONObject) pathJson.get("previewStartingState");
+      if (previewStartingStateJson != null) {
+        previewStartingRotation =
+            Rotation2d.fromDegrees((Double) previewStartingStateJson.get("rotation"));
```

Comment:
Cast this to Number, then get its double value. I've had problems in the past where Java would throw an exception if it gets parsed as an int.

```suggestion
            Rotation2d.fromDegrees(((Number) previewStartingStateJson.get("rotation")).doubleValue());
```